### ISSUE TITLE
Create a more idiomatic way of building a SEP010 Challenge Transaction

### ISSUE THREAD

**Issue by drdgvhbh** [stellar/go#1466]
Creating a challenge transaction for the [SEP010](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md) protocol should be something thats embeded in the SDK, given that most of the parameters can be predetermined.

Current to build such a transaction, you would need to do this...
```go
	randomNounce, err := GenerateRandomString(48)
	if err != nil {
		panic(err)
	}

	var seed [32]byte
	copy(seed[:], []byte("MY_SECRET_KEY_HERE"))
	fiKeyPair, err := keypair.FromRawSeed(seed)
	if err != nil {
		panic(err)
	}
	account.Sequence = "0"
	currentTime := time.Now().UTC().Unix()
	tx := txnbuild.Transaction{
		SourceAccount: &account,
		Timebounds: txnbuild.NewTimebounds(
			currentTime,
			currentTime + 300,
		),
		Operations: []txnbuild.Operation{
			&txnbuild.ManageData{
				SourceAccount: &account,
				Name: "Stellar FI Anchor auth",
				Value: []byte(randomNounce),
			},
		},
		Network: network.TestNetworkPassphrase,
	}
	base64Txn, err := tx.BuildSignEncode(fiKeyPair)
```

Note that I have set the account sequence to `"0"` but the transaction builder will increment it to 1. And you can't set it to -1 because you will get
`couldn't build transaction: failed to parse sequence number: Failed to parse account sequence number: strconv.ParseUint: parsing "-1": invalid syntax`.

Generating a random nounce is trival but still an annoyance. 

Something like the following would be a lot less effort.
```go
	var seed [32]byte
	copy(seed[:], []byte("MY_SECRET_KEY_HERE"))
	fiKeyPair, err := keypair.FromRawSeed(seed)
	if err != nil {
		panic(err)
	}
	tx := txnbuild.Transaction{
		SourceAccount: &account,
		Network: network.TestNetworkPassphrase,
	}
        anchorName := "Stellar FI Anchor auth"
        tx.BuildChallenge(anchorName)
        tx.Sign(fiKeyPair)
	base64Txn, err := tx.Base64()
```

**Comment by ire-and-curses** [comment]
Yes, we need to address at a minimum the "0" problem ASAP, but I really like this convenience function idea as well. @poliha let's hold off on this week's planned release until we can include something to address this, and get it in at the same time. Anchors are implementing SEP 10 so this is urgent.

**Comment by bartekn** [comment]
> Note that I have set the account sequence to "0" but the transaction builder will increment it to 1. 

This is tracked in #1259.

**Comment by poliha** [comment]
There is already a workaround for the "0" problem. Using `Transaction.SimpleAccount` struct you can set `Sequence` to `-1`.  We do something similar for the compliance server, see [here](https://github.com/stellar/go/blob/master/services/compliance/internal/handlers/request_handler_auth_test.go#L108L123). This should help before we implement the convenience function.

**Comment by ire-and-curses** [comment]
@poliha is right. For this edge case (and in general when creating accounts manually), you can create a `SimpleAccount`. It has the following shape:

```
type SimpleAccount struct {
	AccountID string
	Sequence  int64
}
```

Here `Sequence` is an `int64` so can be set to -1 and incremented to 0 in the build step as you needed. The convenience function is still nice but less urgent.

**Comment by poliha** [comment]
Closed in #1468
